src/python/pants/engine/target.py

```diff
@@ -309,6 +299,8 @@ def field_types(self) -> Tuple[Type[Field], ...]:
     def _plugin_field_cls(cls) -> Type:
         # NB: We ensure that each Target subtype has its own `PluginField` class so that
         # registering a plugin field doesn't leak across target types.
+        # This is somewhat similar to scala's path-dependent types:
+        # https://stackoverflow.com/a/2694607/2518889.
```

Comment:
I'm not sure this comment is worth memorializing in source code. While interesting, it might be distracting for people not already familiar with that Scala mechanism.

Reply:
I'm fine with this, but I think it is generally a good thing to refer to existing languages or frameworks that use the patterns we've developed from whole cloth here. Would it be acceptable to try to move this to the docsite?

Reply:
I think it might be more confusing on the docsite, as this is a really specific implementation detail that doesn't need to be mentioned in the docs. Note that this is a private field.

My recommendation is to remove the comment, or perhaps ask others what they think.